### DISCUSSION TITLE
TS/JS cloud functions beta testers

### DISCUSSION THREAD

**Discussion by benallfree** [pockethost/pockethost#76] [Q&A]
If you'd like to be a beta tester for TS/JS cloud functions, please add a note in this thread and follow this thread.

**Comment by yoheisoito** [comment]
Hi! I'd like to participate in the beta testing for the TS/JS cloud functions for pockethost. Thanks!

**Reply by yoheisoito** [reply]
I personally think what Rowy.io does with firebase in terms of receiving cell update triggers and then executing google cloud functions as derivatives to update the triggered document would be a great add on to pockethost and pocketbase all together. One wouldn't go through google's APIs for the case of pocketbase, but if there was an easy way to access other services via javascript for returning the values to update the document where the cell trigger occurred would be awesome, in my view.

**Reply by benallfree** [reply]
Can you provide a more specific use case and some imaginary code samples? I'm not clear on the flow you're proposing.

**Reply by yoheisoito** [reply]
Sorry about that. So upon an update in a listener field (for example text to be translated into another language), run something like below and return the result on async and feed it to a predefined field. So the ability to code using a row.field notation.

```
const derivative: Derivative = async ({ row }) => {
  const { Translate } = require("@google-cloud/translate").v2;
  // Instantiates a client
  const translate = new Translate();
  // Translates text field into Spanish
  const [translation] = await translate.translate(row.text, "es");
  return translation;
};

```

<img width="1222" alt="Screen Shot 2022-11-04 at 0 09 21" src="https://user-images.githubusercontent.com/115172813/199758835-9debaaa2-4029-4026-8740-f00ce4e10d29.png">

**Reply by benallfree** [reply]
Are you saying to do this as a cloud function? In PocketBase, you would implement that as a Go hook, something like this (I'm not an expert):

```go
OnRecordUpdated(func (rec Record) {
   // call google translate in here
   // update rec.DerivedField = translatedValue
})
```

Are you saying you'd like to do something similar as a TS/JS function? A cloud function, something like:

```ts
onRecordUpdated( rec=>{
   rec.translatedValue = await callGoogleTranslate(rec.originalValue)
})
```

**Reply by yoheisoito** [reply]
Yes exactly, I would like to have the ability to write cloud TS/JS functions that would act as backend go hooks, without the need to write in go. My apologies if that was not the objective of the to-be-implemented cloud TS/JS functions for pockethost.

**Reply by benallfree** [reply]
You understand the purpose exactly :)

For your use case, we would need to implement a fetch() function. I’ll add that ticket.

**Reply by yoheisoito** [reply]
very cool! I will definitely look forward for this..!

**Reply by benallfree** [reply]
You understand the purpose exactly :)

For your use case, we would need to implement a fetch() function. I’ll add that ticket.

**Reply by benallfree** [reply]
Track in #80
